plugins/repository-gcs/build.gradle

```diff
@@ -54,6 +54,8 @@ dependencies {
   compile 'com.google.apis:google-api-services-storage:v1-rev20190426-1.28.0'
 
   testCompile project(':test:fixtures:gcs-fixture')
+  // required by the test for the encrypted GCS repository
+  testCompile project(path: ':x-pack:plugin:repository-encrypted', configuration: 'testArtifacts')
```

Comment:
I'm not the right person to judge this but this pattern in the build seems troubling to me. We are using non-Apache licensed code in the tests of Apache licensed code now. Is that ok? Maybe it's better to run these tests in a separate module downstream?
You are definitely using this dependency in Apache licensed files, this seems wrong.

Reply:
_we discussed this synchronously, and also the tests requiring this might not be relevant anymore, but I'm presenting my counter for posterity_

I see your point, and I didn't look at it like this. I had minor qualms about having tests for non-Apache licensed code as Apache licensed. The fact that we pull non-Apache test artifacts is a slightly different point.
But, I tend to dismiss the Apache purists that get offended if the tests require an artifact dependency which is not Apache licensed. It feels squarely like a mean-intending exaggeration; If we don't publish the tests jar, they should not even notice.

When I went on this path, I worried that the reverse alternative (having the encrypted repo depend in tests on the cloud repository) would entail more project setup boilerplate code. I'm not so sure about it right now, and I think it is beneficial if we keep these tests inside the encrypted-repo module, for subjective reasons of code organization.